tests/unittests/tests-sock_dns_cache/tests-sock_dns_cache.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include "net/af.h"
+#include "net/ipv6.h"
+#include "ztimer.h"
+
+#include "tests-sock_dns_cache.h"
+#include "../net/application_layer/sock_dns/dns_cache.h"
+
+static void test_dns_cache_add(void)
+{
+    ipv6_addr_t addr_in = IPV6_ADDR_ALL_NODES_IF_LOCAL;
+    ipv6_addr_t addr_out;
+
+    TEST_ASSERT_EQUAL_INT(0, sock_dns_cache_query("example.com", &addr_out, AF_INET6));
+
+    /* add DNS entry, set it to expire in 1s */
+    sock_dns_cache_add("example.com", &addr_in, sizeof(addr_in), 1);
+    TEST_ASSERT_EQUAL_INT(sizeof(addr_out), sock_dns_cache_query("example.com", &addr_out, AF_INET6));
+    TEST_ASSERT_EQUAL_INT(0, memcmp(&addr_in, &addr_out, sizeof(addr_in)));
+
+    TEST_ASSERT_EQUAL_INT(0, sock_dns_cache_query("example.com", &addr_out, AF_INET));
+    TEST_ASSERT_EQUAL_INT(0, sock_dns_cache_query("alt.example.com", &addr_out, AF_INET6));
+    TEST_ASSERT_EQUAL_INT(0, sock_dns_cache_query("example.comm", &addr_out, AF_INET6));
+    TEST_ASSERT_EQUAL_INT(0, sock_dns_cache_query("example.co", &addr_out, AF_INET6));
+
+    ztimer_sleep(ZTIMER_MSEC, 2000);
```

Comment:
Do we really have to have 2 seconds here? Can't you mock the now value or timeout some how to enforce a timeout?

Reply:
Alternatively, for testing you could make the clock resolution of `sock_dns_cache_add()`'s `ttl` configurable, and have it in milliseconds specifically for the tests.